core/src/solution_submission.rs

```diff
@@ -639,4 +652,41 @@ mod tests {
             }
         };
     }
+
+    // Silly way to create a TimeoutError because the type can't be constructed directly.
+    fn timeout_error() -> TimeoutError {
```

Comment:
:rofl:, I've definitely done this with some std library error types that have the same problem. Not super important but you can also make it a `lazy_static!` constant - but that's really nit-picky.